From b94cc507a56e648551069825663da95adbaacada Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 3 Jun 2019 13:40:34 +0800
Subject: [PATCH] refpolicy: fix avc denials for minimum type

Fix:
type=AVC msg=audit(1559546758.146:30): avc:  denied  { nnp_transition }
for  pid=175 comm="(imesyncd)" scontext=system_u:system_r:init_t:s0
tcontext=system_u:system_r:initrc_t:s0 tclass=process2 permissive=1

type=AVC msg=audit(1559546758.226:33): avc:  denied  { write } for
pid=166 comm="systemd-network" name="system_bus_socket" dev="tmpfs"
ino=12823 scontext=system_u:system_r:systemd_networkd_t:s0
tcontext=system_u:object_r:var_run_t:s0 tclass=sock_file permissive=1

type=AVC msg=audit(1559546758.226:34): avc:  denied  { read } for
pid=166 comm="systemd-network" name="system_bus_socket" dev="tmpfs"
ino=12823 scontext=system_u:system_r:systemd_networkd_t:s0
tcontext=system_u:object_r:var_run_t:s0 tclass=sock_file permissive=1

type=AVC msg=audit(1559546758.226:35): avc:  denied  { write } for
pid=184 comm="systemd-resolve" name="system_bus_socket" dev="tmpfs"
ino=12823 scontext=system_u:system_r:systemd_resolved_t:s0
tcontext=system_u:object_r:var_run_t:s0 tclass=sock_file permissive=1

type=AVC msg=audit(1559546758.226:36): avc:  denied  { read } for
pid=184 comm="systemd-resolve" name="system_bus_socket" dev="tmpfs"
ino=12823 scontext=system_u:system_r:systemd_resolved_t:s0
tcontext=system_u:object_r:var_run_t:s0 tclass=sock_file permissive=1

type=AVC msg=audit(1559546758.227:37): avc:  denied  { connectto } for
pid=166 comm="systemd-network" path="/run/dbus/system_bus_socket"
scontext=system_u:system_r:systemd_networkd_t:s0
tcontext=system_u:system_r:initrc_t:s0 tclass=unix_stream_socket
permissive=1

type=AVC msg=audit(1559546758.234:38): avc:  denied  { connectto } for
pid=184 comm="systemd-resolve" path="/run/dbus/system_bus_socket"
scontext=system_u:system_r:systemd_resolved_t:s0
tcontext=system_u:system_r:initrc_t:s0 tclass=unix_stream_socket
permissive=1

type=AVC msg=audit(1559546758.453:45): avc:  denied  { write } for
pid=194 comm="systemd-logind" name="system_bus_socket" dev="tmpfs"
ino=12823 scontext=system_u:system_r:systemd_logind_t:s0
tcontext=system_u:object_r:var_run_t:s0 tclass=sock_file permissive=1

type=AVC msg=audit(1559546758.453:45): avc:  denied  { connectto } for
pid=194 comm="systemd-logind" path="/run/dbus/system_bus_socket"
scontext=system_u:system_r:systemd_logind_t:s0
tcontext=system_u:system_r:initrc_t:s0 tclass=unix_stream_socket
permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/global_tunables              | 2 +-
 policy/modules/system/init.if       | 1 +
 policy/modules/system/locallogin.te | 1 +
 policy/modules/system/systemd.te    | 9 +++++++++
 4 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/policy/global_tunables b/policy/global_tunables
index affc020..e4ae065 100644
--- a/policy/global_tunables
+++ b/policy/global_tunables
@@ -30,7 +30,7 @@ gen_tunable(allow_execmod,false)
 ## Allow unconfined executables to make their stack executable.  This should never, ever be necessary. Probably indicates a badly coded executable, but could indicate an attack. This executable should be reported in bugzilla")
 ## </p>
 ## </desc>
-gen_tunable(allow_execstack,false)
+gen_tunable(allow_execstack,true)
 
 ## <desc>
 ## <p>
diff --git a/policy/modules/system/init.if b/policy/modules/system/init.if
index c971652..9824473 100644
--- a/policy/modules/system/init.if
+++ b/policy/modules/system/init.if
@@ -1533,6 +1533,7 @@ interface(`init_domtrans_script',`
 
 	files_list_etc($1)
 	domtrans_pattern($1, initrc_exec_t, initrc_t)
+	allow $1 initrc_t:process2 { nnp_transition nosuid_transition };
 
 	ifdef(`enable_mcs',`
 		range_transition $1 initrc_exec_t:process s0;
diff --git a/policy/modules/system/locallogin.te b/policy/modules/system/locallogin.te
index 54fd766..eba273e 100644
--- a/policy/modules/system/locallogin.te
+++ b/policy/modules/system/locallogin.te
@@ -314,3 +314,4 @@ allow local_login_t tmpfs_t:file { create open read write lock };
 allow local_login_t init_var_run_t:fifo_file write;
 allow local_login_t initrc_t:dbus send_msg;
 allow initrc_t local_login_t:dbus send_msg;
+write_sock_files_pattern(local_login_t , initrc_var_run_t, initrc_var_run_t)
diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 350fc81..8827eea 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -472,6 +472,9 @@ allow systemd_logind_t systemd_sessions_var_run_t:dir manage_dir_perms;
 allow systemd_logind_t systemd_sessions_var_run_t:file manage_file_perms;
 allow systemd_logind_t systemd_sessions_var_run_t:fifo_file manage_fifo_file_perms;
 
+allow systemd_logind_t initrc_t:unix_stream_socket connectto;
+rw_sock_files_pattern(systemd_logind_t, var_run_t, var_run_t)
+
 kernel_read_kernel_sysctls(systemd_logind_t)
 
 dev_getattr_dri_dev(systemd_logind_t)
@@ -716,6 +719,9 @@ allow systemd_networkd_t self:udp_socket create_socket_perms;
 allow systemd_networkd_t self:unix_dgram_socket create_socket_perms;
 allow systemd_networkd_t self:netlink_generic_socket create_socket_perms;
 
+allow systemd_networkd_t initrc_t:unix_stream_socket connectto;
+rw_sock_files_pattern(systemd_networkd_t, var_run_t, var_run_t)
+
 manage_dirs_pattern(systemd_networkd_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
 manage_files_pattern(systemd_networkd_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
 manage_lnk_files_pattern(systemd_networkd_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
@@ -1029,6 +1035,9 @@ allow systemd_resolved_t self:process { getcap setcap setfscreate signal };
 
 allow systemd_resolved_t self:tcp_socket { accept listen };
 
+allow systemd_resolved_t initrc_t:unix_stream_socket connectto;
+rw_sock_files_pattern(systemd_resolved_t, var_run_t, var_run_t)
+
 manage_dirs_pattern(systemd_resolved_t, systemd_resolved_var_run_t, systemd_resolved_var_run_t)
 manage_files_pattern(systemd_resolved_t, systemd_resolved_var_run_t, systemd_resolved_var_run_t)
 init_pid_filetrans(systemd_resolved_t, systemd_resolved_var_run_t, dir)
-- 
2.7.4

